### PR TITLE
Address go-report issues.

### PR DESCRIPTION
**Pull request by boumenot**
Unused variable.
Typo.

### CODE REVIEW

File: autorest/azure/token_test.go [hunk 1]
```diff
@@ -496,7 +496,7 @@ func newServicePrincipalTokenCertificate() *ServicePrincipalToken {
 	if err != nil {
 		panic(err)
 	}
-	certificate, err := x509.ParseCertificate(certificateBytes)
+	certificate, _ := x509.ParseCertificate(certificateBytes)
```
Comment (ahmetalpbalkan) [inline autorest/azure/token_test.go]: please check the error instead

**Comment by ahmetalpbalkan** [comment]
LGTM except nits

File: autorest/azure/token_test.go (hunk 1 above)
Comment (boumenot) [inline autorest/azure/token_test.go]: I made the code panic, just as it does above.

**Comment by ahmetalpbalkan** [comment]
@boumenot ugh whoever put these panics in there, not a good idea. test helpers can take an instance of `*testing.T` and just call `t.Fatal` when things go wrong

**Comment by colemickens** [comment]
That would be me. I'm assuming panic() actually stops the tests in their tracks?

I can send a PR changing the `panic`s to `t.Fatal`s.

**Comment by colemickens** [comment]
https://github.com/Azure/go-autorest/pull/70

**Pull request merged**